src/tests/Interop/MarshalAPI/FunctionPointer/FunctionPointer.cs

```diff
@@ -111,4 +112,81 @@ public static void RunGetDelForOutIntTest()
         }
         Assert.Equal(expectedValue, outVar);
     }
+
+    [UnmanagedCallersOnly(CallConvs = [typeof(CallConvCdecl)])]
+    static int UnmanagedExportedFunction(float arg)
+    {
+        return Convert.ToInt32(arg);
+    }
+
+    class GenericCaller<T>
+    {
+        internal static unsafe T GenericCalli<U>(void* fnptr, U arg)
+        {
+            return ((delegate* unmanaged[Cdecl]<U, T>)fnptr)(arg);
+        }
+
+        internal static unsafe BlittableGeneric<T> WrappedGenericCalli<U>(void* fnptr, U arg)
+        {
+            return ((delegate* unmanaged[Cdecl]<U, BlittableGeneric<T>>)fnptr)(arg);
```

Comment:
There is calling convention mismatch. `int` and `struct { int }` are not always treated the same in the ABI. This needs to avoid casting between incompatible signatures.